core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/HealthMonitor.scala

```diff
@@ -0,0 +1,238 @@
+package org.broadinstitute.dsde.rawls.monitor
+
+import java.util.concurrent.TimeoutException
+
+import akka.actor.{Actor, Props}
+import akka.pattern.{after, pipe}
+import cats._
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, MethodRepoDAO, SlickDataSource, UserDirectoryDAO}
+import org.broadinstitute.dsde.rawls.google.GooglePubSubDAO
+import org.broadinstitute.dsde.rawls.model.Subsystems._
+import org.broadinstitute.dsde.rawls.model.{StatusCheckResponse, SubsystemStatus}
+import org.broadinstitute.dsde.rawls.monitor.HealthMonitor._
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.util.control.NonFatal
+
+/**
+  * Created by rtitle on 5/17/17.
+  */
+object HealthMonitor {
+  val DefaultFutureTimeout = 1 minute
+  val DefaultStaleThreshold = 15 minutes
+
+  val OkStatus = SubsystemStatus(true, None)
+  val UnknownStatus = SubsystemStatus(false, Some(List("Unknown status")))
+  def failedStatus(message: String) = SubsystemStatus(false, Some(List(message)))
+
+  // Actor API:
+
+  /** Triggers subsystem checking */
+  case object CheckAll
+  /** Stores status for a particular subsystem */
+  case class Store(subsystem: Subsystem, status: SubsystemStatus)
+  /** Retrieves current status and sends back to caller */
+  case object GetCurrentStatus
+
+  def props(slickDataSource: SlickDataSource, googleServicesDAO: GoogleServicesDAO, googlePubSubDAO: GooglePubSubDAO, userDirectoryDAO: UserDirectoryDAO, methodRepoDAO: MethodRepoDAO,
+            groupsToCheck: Seq[String], topicsToCheck: Seq[String], bucketsToCheck: Seq[String],
+            futureTimeout: FiniteDuration = DefaultFutureTimeout, staleThreshold: FiniteDuration = DefaultStaleThreshold): Props =
+    Props(new HealthMonitor(slickDataSource, googleServicesDAO, googlePubSubDAO, userDirectoryDAO, methodRepoDAO, groupsToCheck, topicsToCheck, bucketsToCheck, futureTimeout, staleThreshold))
+}
+
+class HealthMonitor private (val slickDataSource: SlickDataSource, val googleServicesDAO: GoogleServicesDAO, val googlePubSubDAO: GooglePubSubDAO, val userDirectoryDAO: UserDirectoryDAO, val methodRepoDAO: MethodRepoDAO,
+                             val groupsToCheck: Seq[String], val topicsToCheck: Seq[String], val bucketsToCheck: Seq[String],
+                             val futureTimeout: FiniteDuration, val staleThreshold: FiniteDuration) extends Actor with LazyLogging {
+  // Use the execution context for this actor's dispatcher for all asynchronous operations.
+  // We define a separate execution context (a fixed thread pool) for health checking to
+  // not interfere with user facing operations.
+  import context.dispatcher
+
+  /**
+    * Contains each subsystem status along with a timestamp of when the entry was made.
+    * Initialized with unknown status.
+    */
+  var data: Map[Subsystem, (SubsystemStatus, Long)] = {
```

Comment:
You should make it private though even though technically no one should be able to get at it since it's all actorrefs. Over in cromwell we generally prefer FSM to context.become but it's all basically the same.